Mock of first Profile component

jonquag opened the pull request:
Added grid functionality and styled elements.

Todo: Fix fontFamily and a bug with e-mail running loose.

jonquag committed:
Mock of first Profile component

Added grad functionality and styled elements.

Todo: Fix fontFamily and a bug with e-mail running loose.

jonquag committed:
Fixed bug with alignment on grid

Fixed a bug that was distorting alignment. Added responsiveness to xs screens.

aeciorc commented on client/src/pages/Profile2.jsx:
```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import TextField from 'material-ui/TextField';
```
This should be ` "@material-ui/core/TextField";`

aeciorc commented on client/src/pages/Profile2.jsx:
```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
```
You don't need a theme provider here

jonquag committed:
Updated hooks

Went over my code with Amha. This is the updated version working with hooks and cleaner code.

jonquag committed:
Added routing functionality

Routing now works within the Profile page.

Todo: add styling

jessdesr commented on client/src/components/ProfileUserInfo.js:
```diff
@@ -0,0 +1,120 @@
+                </Grid>
+                {/* First Name  */}
+                <Grid item xs={12} sm={6}>
+                    <FormHelperText variant='h6'>First Name</FormHelperText>{' '}
```
You may want to use InputLabel here instead of FormHelperText

jessdesr commented on client/src/components/ProfileUserInfo.js:
```diff
@@ -0,0 +1,120 @@
+    const classes = useStyles();
+
+    return (
+        <React.Fragment>
```
Wrapping everything in a React Fragment isn't necessary since you've got the Grid parent component.  Just move your debug stuff inside the Grid component or delete it.

jessdesr commented on client/src/components/ProfileSettings.js:
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Grid } from '@material-ui/core';
+
+const ProfileSettings = () => {
```
Organizational suggestion: I would move all of these components into `components/Profile` and remove Profile from all of these component names.

jessdesr commented on client/src/components/ProfileSidebar.js:
```diff
@@ -0,0 +1,25 @@
+            </Grid>
+            {/* Navbar container */}
+            <Grid container item xs={12} direction='column'>
+                <Link to='/profile/'>User Info</Link>
```
Change to Navlink https://reactrouter.com/web/api/NavLink

jessdesr commented on client/src/pages/Landing.js:
```diff
@@ -1,71 +1,18 @@
+
+    render() {
+        return (
+            <React.Fragment>
```
Similarly, no need to wrap in React Fragment when there's only one root component.

jessdesr commented on client/src/components/ProfileSidebar.js:
```diff
@@ -0,0 +1,25 @@
+const ProfileSidebar = () => {
+    return (
+        // Sidebar container
+        <Grid container>
```
Take a look at the Drawer component to make this sidebar: https://material-ui.com/components/drawers/

jessdesr reviewed (commented):
Looking good so far structurally!

jessdesr commented:
Just a general thought for front-end pull requests: 
- If you wouldn't mind throwing up a screenshot of the page, it'll let me comment on styling of the page without needing to run the app myself.

jonquag commented:
![user-profile-component](https://user-images.githubusercontent.com/67391359/104641687-19921380-5678-11eb-821b-4437e1f929c8.png)
Here's a screenshot of the current ui look.

jonquag committed:
Testing Drawer

Can't fill remaining space in Drawer component

jonquag committed:
Working with Drawer

I've addded the drawer and am trying to debug some styling.

jonquag commented:
![profile-drawer](https://user-images.githubusercontent.com/67391359/104747771-e5256280-571e-11eb-9406-6750769f78c0.png)

jonquag commented:
I'm having trouble setting the width of the drawer with styling. However, I can affect the relative width by adding content to the drawer.

I've been using a video from the videos Aecio sent earlier in the week to try to solve this issue and can't quite put a finger on why I can't style it.

https://www.youtube.com/watch?v=CjFWbEOcq-Y&list=PLQg6GaokU5CwiVmsZ0d_9Zsg_DnIP_xwr&index=21&t=967s

jonquag committed:
Fixed drawer stylings

Centered profile pic and name.
Set color to drawer.

todo: fix UserInfo component spacing.

jonquag commented:
![profile-drawer2](https://user-images.githubusercontent.com/67391359/104755738-d774da80-5728-11eb-850c-56755e2ae5f8.png)


Here is an updated look

aeciorc reviewed (commented):
Looks good Jon! The the avatar is a little stretched out, you'll want to change the CSS so that it maintains the correct ratio. Besides that, you'll want to integrate the changes from dev into this and then convert it to a regular PR. Then we can approve and merge it

jonquag committed:
Merge branch 'dev' into FE-Onboarding-Profile-#3

jonquag committed:
Updated Profile component

jonquag committed:
Fixed routing issue

/profile/ now automatically loads the UserInfo component.

jessdesr commented on client/src/App.js:
```diff
@@ -6,6 +6,7 @@ import { theme } from './themes/theme';
 // import LandingPage from './pages/Landing';
 import Signup from './pages/auth/Sign-up';
 import Login from './pages/auth/Login';
+// import UserInfo from './components/Profile/UserInfo';
```
You can delete the commented code: if you need it back later on, you can just add it back in!

jessdesr commented on client/src/App.js:
```diff
@@ -17,8 +18,8 @@ function App() {
                         <Route exact path="/login" component={Login} />
+
                         <Route exact path="/sign-up" component={Signup} />
 
```
Add a route for "/profile"

jessdesr commented on client/src/components/Profile/Courses.js:
```diff
@@ -0,0 +1,13 @@
+import { Grid } from '@material-ui/core';
+
+const Courses = () => {
+    return (
```
We probably don't need this template component now

jessdesr commented on client/src/components/Profile/Drawer.js:
```diff
@@ -0,0 +1,111 @@
+    },
+}));
+
+const Drawer = props => {
```
I might rename this to Sidebar

jessdesr commented on client/src/components/Profile/Drawer.js:
```diff
@@ -0,0 +1,111 @@
+import { NavLink } from 'react-router-dom';
+import { makeStyles } from '@material-ui/styles';
+import profilePic from '../../static/images/profilePicSample.png';
+// import Avatar from '@material-ui/core/Avatar';
```
Delete commented code

jessdesr commented on client/src/components/Profile/Drawer.js:
```diff
@@ -0,0 +1,111 @@
+                container
+                direction="column"
+                alignItems="center"
+                style={{ width: '400px' }}
```
I would avoid combining inline styling with makeStyles.  Move this to the `test` class (maybe rename `test` to be less.. testy 😆 )

jessdesr commented on client/src/components/Profile/Drawer.js:
```diff
@@ -0,0 +1,111 @@
+                    </NavLink>
+                </Grid>
+                <Grid container direction="column" item justify="flex-end" sm={4}>
+                    <Typography align="left">Logout</Typography>
```
This will need to be changed into a button when this is being integrated with the backend

jonquag replied:
I also had a little trouble with aligning this to the left as shown in the .sketch file.

jessdesr commented on client/src/pages/Landing.js:
```diff
@@ -1,71 +1,13 @@
-import React, { Component } from "react";
+import React, { Component } from 'react';
```
We can probably delete this file now as well

jessdesr reviewed (commented):
I think this is ready to merge after the cleanup notes I mentioned.

jonquag committed:
Deleted unused code

Deleted unused coments and components. Renamed Drawer to Sidebar. Touched up some styling code,
